feature/platform/transceiver/tests/zr_firmware_version_test/zr_firmware_version_test.go

```diff
@@ -0,0 +1,161 @@
+	d := &oc.Root{}
+	i := d.GetOrCreateInterface(dp.Name())
+	i.Enabled = ygot.Bool(true)
+	i.Type = oc.IETFInterfaces_InterfaceType_ethernetCsmacd
+	gnmi.Replace(t, dut1, gnmi.OC().Interface(dp.Name()).Config(), i)
+	OCcomponent := opticalChannelComponentFromPort(t, dut1, dp)
+	gnmi.Replace(t, dut1, gnmi.OC().Component(OCcomponent).OpticalChannel().Config(), &oc.Component_OpticalChannel{
+		TargetOutputPower: ygot.Float64(targetOutputPower),
+		Frequency:         ygot.Uint64(frequency),
+	})
+}
+func verifyFirmwareVersionValue(t *testing.T, dut1 *ondatra.DUTDevice, pStream *samplestream.SampleStream[string]) string {
+	firmwareVersionSample := pStream.Next(t)
+	if firmwareVersionSample == nil {
+		t.Fatalf("Firmware telemetry %v was not streamed in the most recent subscription interval", firmwareVersionSample)
+	}
+	firmwareVersionVal, ok := firmwareVersionSample.Val()
+	if !ok {
+		t.Fatalf("Firmware version %q telemetry is not present", firmwareVersionSample)
+	}
+	// Check firmware version return value of correct type
+	if reflect.TypeOf(firmwareVersionVal).Kind() != reflect.String {
+		t.Fatalf("Return value is not type string")
+	}
+	t.Logf("Firmware version sample value: %v", firmwareVersionVal)
+	return firmwareVersionVal
+}
+
+func TestZRFirmwareVersionState(t *testing.T) {
+	dut1 := ondatra.DUT(t, "dut")
+	dp1 := dut1.Port(t, "port1")
+	dp2 := dut1.Port(t, "port2")
+	t.Logf("dut1: %v", dut1)
+	t.Logf("dut1 dp1 name: %v", dp1.Name())
+	intUpdateTime := time.Minute
+	interfaceConfig(t, dut1, dp1)
+	interfaceConfig(t, dut1, dp2)
+	gnmi.Await(t, dut1, gnmi.OC().Interface(dp1.Name()).OperStatus().State(), intUpdateTime, oc.Interface_OperStatus_UP)
+	transceiverName := gnmi.Get(t, dut1, gnmi.OC().Interface(dp1.Name()).Transceiver().State())
+	// Check if TRANSCEIVER is of type 400ZR
+	if dp1.PMD() != ondatra.PMD400GBASEZR {
+		t.Fatalf("%s Transceiver is not 400ZR its of type: %v", transceiverName, dp1.PMD())
+	}
+	component1 := gnmi.OC().Component(transceiverName)
+
+	p1Stream := samplestream.New(t, dut1, component1.FirmwareVersion().State(), 10*time.Second)
+
+	firmwareVersion := verifyFirmwareVersionValue(t, dut1, p1Stream)
+	t.Logf("Port1 dut1 %s Firmware version: %v", dp1.Name(), firmwareVersion)
+
+	p1Stream.Close()
+}
+func TestZRFirmwareVersionStateInterfaceFlap(t *testing.T) {
+	dut1 := ondatra.DUT(t, "dut")
+	dp1 := dut1.Port(t, "port1")
+	dp2 := dut1.Port(t, "port2")
+	t.Logf("dut1: %v", dut1)
+	t.Logf("dut1 dp1 name: %v", dp1.Name())
+	interfaceConfig(t, dut1, dp1)
+	interfaceConfig(t, dut1, dp2)
+	intUpdateTime := time.Minute
+	gnmi.Await(t, dut1, gnmi.OC().Interface(dp1.Name()).OperStatus().State(), intUpdateTime, oc.Interface_OperStatus_UP)
+	transceiverName := gnmi.Get(t, dut1, gnmi.OC().Interface(dp1.Name()).Transceiver().State())
+	// Check if TRANSCEIVER is of type 400ZR
+	if dp1.PMD() != ondatra.PMD400GBASEZR {
+		t.Fatalf("%s Transceiver is not 400ZR its of type: %v", transceiverName, dp1.PMD())
+	}
+	// Disable interface
```

Comment:
Can you add an argument for configInterface() for the enabled state? Then you can just call configInterface() here instead of these lines.